ironmon_tracker/network/EventData.lua

```diff
@@ -0,0 +1,1196 @@
+EventData = {}
+
+-- Helper Functions and Variables
+
+---Searches for a Pokémon by name, finds the best match; returns 0 if no good match
+---@param name string?
+---@param threshold number? Default threshold distance of 3
+---@return number pokemonId
+local function findPokemonId(name, threshold)
+	if name == nil or name == "" then
+		return 0
+	end
+	threshold = threshold or 3
+	-- Format list of Pokemon as id, name pairs
+	local pokemonNames = {}
```

Comment:
Rebuilding the tables like this every call may affect performance? Will have to do some testing. If it does, it's very simple to just rebuild the tables outside of these "find" functions